### PR TITLE
Update members tangle calculation for exclusion spec

### PR DESCRIPTION
**Pull request by Powersource**
Fixes https://github.com/ssbc/ssb-tribes2/issues/69

Based on https://github.com/ssbc/ssb-tribes2/pull/65

### CODE REVIEW

**Comment by Powersource** [comment]
Was first thinking I could get the epoch by getting the latest epoch tangle and using the `previous`. But that might be an array. More accurate to look it up based on the current writeKey.

**Comment by Powersource** [comment]
Hmm I can't just get my own feed with the secret as the purpose though since someone else might've been the creator of the epoch init.

**Comment by Powersource** [comment]
Tested performance using the `get-tangle-100-publishes` test

With group, and epoch tangles:
```
getTangles: 17.845s
```

With group, epoch, and members tangles:
```
getTangles: 20.140s
```

**Comment by Powersource** [comment]
With only the group tangle it was
```
getTangles: 16.390s
```

File: lib/get-tangle.js [hunk 1]
```diff
@@ -89,11 +126,9 @@ function GetUpdates(ssb, tangle) {
...
-    pull(
-      ssb.db.query(where(and(tangleRoot(id))), toPullStream()),
-      pull.collect(cb)
-    )
-  }
+  const { where, and, toPullStream } = ssb.db.operators
+  pull(
+    ssb.db.query(where(and(tangleRoot(root))), toPullStream()),
+    pull.collect(cb)
+  )
```
Comment (mixmix) [inline lib/get-tangle.js:133]:
```suggestion
  const { where, toCallback } = ssb.db.operators
  ssb.db.query(where(tangleRoot(root)), toCallback(cb))
```

File: test/get-tangle.test.js [hunk 2]
```diff
@@ -110,13 +118,16 @@ test(`get-tangle-${n}-publishes`, (t) => {
     t.error(err, 'no error creating group')
 
     const groupId = data.id
+
+    //console.time('getTangles')
+
```
Comment (mixmix) [inline test/get-tangle.test.js:123]:
```suggestion
```

File: test/get-tangle.test.js [hunk 3]
```diff
@@ -129,6 +140,8 @@ test(`get-tangle-${n}-publishes`, (t) => {
         (err) => {
           t.error(err, 'no error publishing')
 
+          //console.timeEnd('getTangles')
```
Comment (mixmix) [inline test/get-tangle.test.js:143]:
```suggestion
```

File: lib/get-tangle.js [hunk 4]
```diff
@@ -17,57 +16,95 @@ function toUri(link) {
...
-      return cb(new Error(`get-tangle expects valid groupId, got: ${groupId}`))
+    if (!info) {
+      return cb(new Error(`get-tangle: unknown groupId ${groupId}`))
     }
 
-    server.box2.getGroupInfo(groupId, (err, info) => {
+    if (tangle === 'members') {
+      // we find the id of the init msg for the current epoch
+      pull(
+        server.metafeeds.branchStream({ old: true, live: false }),
```
[the remark this answers is not in the document]
Comment (Powersource) [reply lib/get-tangle.js:31]: I think you have to specify `opts.root` to only get from one person

File: lib/get-tangle.js [hunk 5]
```diff
@@ -17,57 +16,95 @@ function toUri(link) {
...
+  })
+}
+
+/** for figuring out what "previous" should be for the group. `server` is the ssb server you're using. `tangle` is the name of the tangle in the group you're looking for, e.g. "group" or "members" */
+function getTangle(server, tangle, groupId, cb) {
+  if (!isIdentityGroupSSBURI(groupId)) {
+    // prettier-ignore
+    return cb(new Error(`get-tangle expects valid groupId, got: ${groupId}`))
+  }
+
```
[the remark this answers is not in the document]
Comment (Powersource) [reply lib/get-tangle.js:70]: The jitdb cache seems really fast after the first query

File: lib/add-tangles.js [hunk 6]
```diff
@@ -35,20 +33,15 @@ module.exports = function AddTangles(server) {
...
+  if (!content.recps) return cb(new Error('Missing recps when adding tangles'))
 
-    if (!isIdentityGroupSSBURI(content.recps[0])) return cb(null, content)
+  if (!isIdentityGroupSSBURI(content.recps[0]))
+    return cb(new Error('recps[0] is not a group id when adding tangles'))
 
-    addSomeTangles(content, tangles, (err, content) => {
-      // prettier-ignore
-      if (err) return cb(clarify(err, 'failed to add tangles to content'))
+  addSomeTangles(content, tangles, (err, content) => {
```
[the remark this answers is not in the document]
Comment (staltz) [reply lib/add-tangles.js:41]: And `new Error('CLARIFICATION', {cause: err})` is the new `clarify(err, 'CLARIFICATION')`

**Pull request merged**